Update loadData.R

tumorscholar opened the pull request:
this script uses the new cell ranger output. 

I have read the files in the directory
remove the not citeseq and duplicated samples
then used the pattern to get sample ID
then created obj in a loop.

jamesboot commented:
Hi Raju,

A couple of issues you need to correct before we merge:

**ISSUE 1:**

On line 16 you remove the file paths you don't want to load:

```
files <- files[-c(10,16,18,19,28,30,32)]
```

The `samples` vector is made from the `files` vector, therefore when defining the for loop on line 74 you should not remove the samples from the `sample` vector - because they are already not there.

**ISSUE 2:**

The multiple `pattern1`, `pattern2` etc.  and `gsub` commands is a bit clunky, and will get out of hand the more and more samples you add over time. You're having to do this because the folder structure in which you are keeping the data to be loaded is inconsistent. For example, most samples are here:
```
 "/data/Blizard-AlazawiLab/rk/cellranger_counts_rk/GC-WL-9999-SAT-VAT-PBMC/outs/filtered_feature_bc_matrix.h5"  
```

But then the new samples are here:
```
"/data/Blizard-AlazawiLab/rk/cellranger_counts_rk/new_samples/Analysis/CellRanger_Total/GC-WL-10742-LIVER/outs/per_sample_outs/GC-WL-10742-LIVER/count/sample_filtered_feature_bc_matrix.h5"
```
or here:
```
"/data/Blizard-AlazawiLab/rk/cellranger_counts_rk/new_samples/Analysis/Seurat_Demux/h5/GC-WL-10634-SAT-VAT-PBMC_sample_filtered_feature_bc_matrix.h5" 
```
I would make the folder structure more consistent. I think it would be best to put all the cellranger output folders (including the latest samples) in the same folder  e.g. in `/data/Blizard-AlazawiLab/rk/cellranger_counts_rk/`. This should mean that your folder structure will be:
```
 "/data/Blizard-AlazawiLab/rk/cellranger_counts_rk/<SAMPLE_NAME>/outs/filtered_feature_bc_matrix.h5"
```
Then you can just have the following `gsub` commands instead of the multiple arguments you currently have:
```
# Extract sample names from file names
pattern1 <- '/data/Blizard-AlazawiLab/rk/cellranger_counts_rk/'
samples <- gsub(pattern1,
                '',
                files)

pattern2 <- '/outs/filtered_feature_bc_matrix.h5'
samples <- gsub(pattern2,
                '',
                samples)
```

Make the above changes, test the script and then we can merge.

jamesboot commented:
Hi Raju,

Still a couple of issues:

**ISSUE 1:**

You still need line 16:

`files <- files[-c(10,16,18,19,28,30,32)]`

It's only the bit when defining the for loop you need to remove.

**ISSUE 2:**

You will still need a pattern 2 to remove to create the sample names. Pattern 1 should be the directory root i.e. `'/data/Blizard-AlazawiLab/rk/cellranger/'` and pattern 2 will need to be `'/outs/filtered_feature_bc_matrix.h5'`

Let me know when you've made these changes.

tumorscholar commented:
Hello James,
I have moved extra samples to another folder and added pattern 2  in the
script., maybe I have added it in the wrong place

[image: image.png]

On Mon, Jun 3, 2024 at 1:06 PM James Boot ***@***.***> wrote:

> Hi Raju,
>
> Still a couple of issues:
>
> *ISSUE 1:*
>
> You still need line 16:
>
> files <- files[-c(10,16,18,19,28,30,32)]
>
> It's only the bit when defining the for loop you need to remove.
>
> *ISSUE 2:*
>
> You will still need a pattern 2 to remove to create the sample names.
> Pattern 1 should be the directory root i.e.
> '/data/Blizard-AlazawiLab/rk/cellranger/' and pattern 2 will need to be
> '/outs/filtered_feature_bc_matrix.h5'
>
> Let me know when you've made these changes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tumorscholar/MASLD_scMultiome/pull/1#issuecomment-2145025854>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFJUUTS3TIN4ZHCCZKAZ3HDZFRL55AVCNFSM6AAAAABIPHYTLKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNBVGAZDKOBVGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
